### ISSUE TITLE
When setting batchsize too small, the IOU is very low

### ISSUE THREAD

**Issue by facetohard** [CSAILVision/semantic-segmentation-pytorch#54]
I use `UserScatteredDataParallel` and `patch_replication_callback` in my  code, but, when i set the batchsize small (batchsize=2 per GPU), the IOU is very low and as enlarging the batchsize, the IOU is increasing.
I think this result is unusual since the results using `DataParallel` is better in the same batchsize.

**Comment by lxtGH** [comment]
I got the same result with you, use the encoder res101 , batch_size_per_gpu=1 ,the IOU is only 0.212

**Comment by Tete-Xiao** [comment]
@facetohard If you are using 8 GPUs, setting batch_size as 2 per gpu means that the actual batch_size is 2 * 8 = 16, which is the same as our setting. Using DataParallel results in an even smaller batch size (for bn), the results should be worse. Therefore it doesn't make any sense.

I suppose you should check whether you've forgot to set model.eval() during testing. What you have mentioned makes me believe that something's misused in regarding to bn statistics.

**Comment by facetohard** [comment]
@Tete-Xiao I used the training code with normal pytorch training process but changed
`DataParallel` to `UserScatteredDataParallel`, pytorch BN to `SynchronizedBatchNorm2d` . Since the return of net model, which  used in example, is loss and score, I calculated loss and score in net module, and then computed mean loss and did backward in the main loop.
Does this process is right?

**Comment by wuyi1983** [comment]
I have similar problems.

My machine has 4 1080Ti so I could only set batch_size_per_gpu  to 1.

The evaluation performance is
Mean IoU: 0.3685, Accuracy: 77.85%

This is lower than the reference one for ResNet-50_dilated8 + ppm_bilinear_deepsup:
Mean IoU: 0.4126, Accuracy: 79.73%

Does my result make sense?
